CefSharp.Core/Internals/ClientAdapter.cpp

```diff
@@ -994,31 +994,66 @@ namespace CefSharp
             auto argList = message->GetArgumentList();
             IJavascriptCallbackFactory^ callbackFactory = _browserAdapter->JavascriptCallbackFactory;
 
-            if (name == kEvaluateJavascriptResponse || name == kJavascriptCallbackResponse)
+            if (name == kOnFocusedNodeChanged)
             {
-                auto success = argList->GetBool(0);
-                auto callbackId = GetInt64(argList, 1);
-
-                auto pendingTask = _pendingTaskRepository->RemovePendingTask(callbackId);
-                if (pendingTask != nullptr)
+                auto handler = _browserControl->RequestHandler;
+                if (handler != nullptr)
                 {
-                    auto response = gcnew JavascriptResponse();
-                    response->Success = success;
-
-                    if (success)
+                    // 0: frame ID (int 64)
+                    // 1: is a node (bool)
+                    // 2: tag name (string)
+                    // 3: attributes (dictionary)
+                    CefFrameWrapper frameWrapper(browser->GetFrame(GetInt64(argList, 0)));
+                    auto notEmpty = argList->GetBool(1);
+                    if (notEmpty)
                     {
-                        response->Result = DeserializeObject(argList, 2, callbackFactory);
+                        // Node information was passed from the render process.
+                        auto argAttributes = argList->GetDictionary(3);
+                        auto attributes = gcnew System::Collections::Generic::Dictionary<String^, String^>();
+                        CefDictionaryValue::KeyList keys;
+                        argAttributes->GetKeys(keys);
+                        for (auto key : keys)
+                        {
+                            attributes->Add(gcnew String(key.c_str()), gcnew String(argAttributes->GetString(key).c_str()));
+                        }
+
+                        auto node = gcnew DomNode(attributes);
+                        node->TagName = gcnew String(argList->GetString(2).c_str());
```

Comment:
Please pass in the `TagName` through the constructor and make the getter private.